### PR TITLE
Improve docs part 2

### PR DESCRIPTION
**Pull request by aboedo**
Part 2 of adding docstrings to public methods in the Purchases class. 

Public members of other public classes don't have docstrings yet, but I might add them later on after having released this.

### CODE REVIEW

File: RevenueCat/Scripts/Purchases.cs [hunk 1]
```diff
@@ -8,31 +8,6 @@
 
 public partial class Purchases : MonoBehaviour
 {
-    public delegate void LogInFunc(CustomerInfo customerInfo, bool created, Error error);
-
-    public delegate void GetOfferingsFunc(Offerings offerings, Error error);
-
-    public delegate void CheckTrialOrIntroductoryPriceEligibilityFunc(Dictionary<string, IntroEligibility> products);
```
Comment (aboedo) [inline RevenueCat/Scripts/Purchases.cs:15]: I moved these closer to where they're actually used, so it's much easier to understand what they actually do

File: RevenueCat/Scripts/Purchases.cs [hunk 2]
```diff
@@ -499,40 +600,106 @@ public void SyncPurchases()
...
 
+    ///
+    /// <summary>
+    /// Subscriber attributes are useful for storing additional, structured information on a user.
+    /// Since attributes are writable using a public key they should not be used for
+    /// managing secure or sensitive information such as subscription status, coins, etc.
+    /// </summary>
+    /// 
+    /// <remarks>Key names starting with "$" are reserved names used by RevenueCat. For a full list of key
+    /// restrictions refer [to our guide](https://docs.revenuecat.com/docs/subscriber-attributes)
```
Comment (aboedo) [inline RevenueCat/Scripts/Purchases.cs:699]: most of these docstrings are adapted from iOS

**Comment by aboedo** [comment]
@RevenueCat/core-sdk bump

File: RevenueCat/Scripts/Purchases.cs [hunk 3]
```diff
@@ -499,40 +600,106 @@ public void SyncPurchases()
...
+    /// Note: Subscription groups are automatically collected for determining eligibility. If RevenueCat can't
+    /// definitively compute the eligibility, most likely because of missing group information, it will return
+    /// <see cref="IntroEligibilityStatus.IntroEligibilityStatusUnknown"/>.
+    /// The best course of action on unknown status is to display the non-intro
+    /// pricing, to not create a misleading situation. To avoid this, make sure you are testing with the latest
+    /// version of iOS so that the subscription group can be collected by the SDK.
+    /// </remarks>
+    ///
+    /// <param name="products"> The <see cref="StoreProduct"/>  for which you want to compute eligibility.</param>
+    /// <param name="callback"> The <see cref="CheckTrialOrIntroductoryPriceEligibilityCallback"/> callback. </param>
```
Comment (taquitos) [inline RevenueCat/Scripts/Purchases.cs:644]: `CheckTrialOrIntroductoryPriceEligibilityCallback` -> `CheckTrialOrIntroductoryPriceEligibilityFunc`

**Review by taquitos — APPROVED** [review]
One tiny change, but then 📖🖋️🐐
image

File: RevenueCat/Scripts/Purchases.cs (hunk 3 above)
Comment (aboedo) [reply RevenueCat/Scripts/Purchases.cs]: great catch, thanks!!

**Pull request merged**